From 0252d6c4f2f7fef733c9c789afba46aa6d20f350 Mon Sep 17 00:00:00 2001
From: Nirbheek Chauhan <nirbheek@centricular.com>
Date: Fri, 3 Apr 2020 17:21:29 +0530
Subject: [PATCH] meson: Use the `debug` option directly instead of `buildtype`

`get_option('buildtype')` returns `'custom'` in most of the cases
where `-Ddebug` is set. However, `get_option('debug')` will always be
set correctly if the user sets `-Dbuildtype`. See the following table
for the mapping:

https://mesonbuild.com/Builtin-options.html#build-type-options
---
 meson.build | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/meson.build b/meson.build
index 75e796e..65a5719 100644
--- a/meson.build
+++ b/meson.build
@@ -69,8 +69,7 @@ endif
 # This is available pretty much everywhere
 cdata.set('HAVE_STRINGIZE', 1)
 
-buildtype = get_option('buildtype')
-if buildtype == 'debug' or buildtype == 'debugoptimized'
+if get_option('debug')
   cdata.set('DEBUG', 1)
 endif
 
-- 
2.26.0

